### PR TITLE
getUrlForArtifact should check if current user has access to the modeldb entity

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentServiceImpl.java [hunk 1]
```diff
@@ -1128,6 +1128,21 @@ public void getUrlForArtifact(
         throw StatusProto.toStatusRuntimeException(status);
       }
 
+      Map<String, String> projectIdFromExperimentMap =
+          experimentDAO.getProjectIdsByExperimentIds(Collections.singletonList(request.getId()));
+      if (projectIdFromExperimentMap.size() == 0) {
+        errorMessage = "Access is denied. Experiment not found for given id : " + request.getId();
+        ModelDBUtils.logAndThrowError(
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/experiment/ExperimentServiceImpl.java:1135]: this should be a not found? you are not checking for privileges here

**Review by ravishetye — CHANGES_REQUESTED** [review]
IMO needs a change in error code

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentServiceImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/ExperimentServiceImpl.java:1135]: Fixed

**Review by ravishetye — APPROVED** [review]

**Comment by ravishetye** [comment]
@ad-47 : any particular reason this is not merged?

**Pull request merged**